src/utils/mempool.rs

```diff
@@ -182,3 +145,52 @@ impl io::Seek for MemPool {
         io::Seek::seek(&mut self.file, pos)
     }
 }
+
+fn create_shm_fd() -> io::Result<RawFd> {
+    let mut rng = thread_rng();
+
+    loop {
+        match memfd::memfd_create(
+            CStr::from_bytes_with_nul(b"smithay-client-toolkit\0").unwrap(),
+            memfd::MemFdCreateFlag::MFD_CLOEXEC,
+        ) {
+            Ok(fd) => break Ok(fd),
+            Err(nix::Error::Sys(Errno::EINTR)) => continue,
+            Err(nix::Error::Sys(errno)) => break Err(io::Error::from(errno)),
+            Err(_) => {
+                // Fallback to using shm_open
+                let mut mem_file_handle = format!(
+                    "/smithay-client-toolkit-{}",
+                    rng.gen_range(0, ::std::u32::MAX)
+                );
+                break loop {
```

Comment:
I think this block can be flattened like so:

```rust
loop {
    match memfd_create(...) {
        Ok(fd) => return Ok(fd),
        Err(nix::Error::Sys(Errno::EINTR)) => continue,
        Err(nix::Error::Sys(errno)) => return Err(io::Error::from(errno)),
        Err(_) => break,
    }
}
loop {
    /* fallback to using shm_open */
}
```

Reply:
Nice catch